source/common/http/conn_manager_impl.cc

```diff
@@ -1533,19 +1533,39 @@ void ConnectionManagerImpl::ActiveStream::onBelowWriteBufferLowWatermark() {
 }
 
 Tracing::OperationName ConnectionManagerImpl::ActiveStream::operationName() const {
-  return connection_manager_.config_.tracingConfig()->operation_name_;
+  if (connection_manager_.config_.tracingConfig()) {
+    return connection_manager_.config_.tracingConfig()->operation_name_;
+  }
+
+  ENVOY_LOG(error, "tracingConfig()::operationName() called but tracing is not configured");
```

Comment:
For all of these, I would recommend returning an optional, and then letting the caller deal with ignoring or not.

Reply:
@mattklein123 This PR only fixes the crash. It doesn't attempt to change API.

I had done this in this another PR: https://github.com/envoyproxy/envoy/pull/13734. But it went too far in the sense it touched API and became too big. Therefore I thought about going small and only fix the crash issue with this PR and attack the other issue (i.e.: caller knows when tracing is configured or not) in a follow up. What do you think?

Reply:
I would rather just do the proper fix. I don't think it's that bad.